test/re/frida/MethodTest.java

```diff
@@ -39,6 +40,99 @@ public void replacementPropagatesExceptions() {
         thrown.expectMessage("Already dead");
         badger.die();
     }
+    
+    static private String failString = null;
+    static private void Fail( String msg )
+    {
+      failString = msg;
+    }
+    
+    @Test
+    public void TestNewInterface() {
+        loadScript("var X509TrustManager = Java.use('javax.net.ssl.X509TrustManager');" +
+                "try{" +
+                "  var tm = X509TrustManager.$new();" +
+                "}catch(e){" + 
+                "  var MethodTest = Java.use('re.frida.MethodTest');" +
+                "  MethodTest.Fail('couldnt create trustmanager: ' + e);" +
+                "}"
+                );
+        assertNull(failString);
+    }
+    
+    @Test
+    public void TestClassForName() {
+        loadScript("var c = Java.use('java.lang.Class');" +
+                "try{" +
+                "  var orig = c.forName.overload('java.lang.String');" +
+                "  c.forName.overload('java.lang.String').implementation = function(s){ orig(s); };" +
```

Comment:
I've tried adding a `static Class<?>` function to Badger and hooking it.  Frida does not crash in the same way as when hooking `java.lang.Class`.  I also tried hooking some different built-ins with similar returns like `dalvik.system.VMStack.getStackClass2` and even Made a function in Badger that calls through directly to `java.lang.Class.forName()` and hooked that.  So far, I'm unable to reproduce the same crash without directly hooking java.lang.Class.  Maybe we can comment it out for now so it doesn't block the rest of the PR?